packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx

```diff
@@ -1,50 +1,86 @@
-import React, { FC, useContext } from 'react';
+import React, { useState } from 'react';
 import { css } from '@emotion/css';
 import { Modal } from '../Modal/Modal';
 import { IconName } from '../../types/icon';
 import { Button } from '../Button';
-import { stylesFactory, ThemeContext } from '../../themes';
+import { useStyles } from '../../themes';
 import { GrafanaTheme } from '@grafana/data';
-import { HorizontalGroup } from '..';
+import { HorizontalGroup, Input } from '..';
+import { selectors } from '@grafana/e2e-selectors';
 
-export interface Props {
+export interface ConfirmModalProps {
   /** Toggle modal's open/closed state */
   isOpen: boolean;
   /** Title for the modal header */
   title: string;
   /** Modal content */
   body: React.ReactNode;
+  /** Modal description */
+  description?: React.ReactNode;
   /** Text for confirm button */
   confirmText: string;
   /** Text for dismiss button */
   dismissText?: string;
   /** Icon for the modal header */
   icon?: IconName;
+  /** Text user needs to fill in before confirming */
+  confirmationText?: string;
+  /** Text for alternative button */
+  alternativeText?: string;
   /** Confirm action callback */
   onConfirm(): void;
   /** Dismiss action callback */
   onDismiss(): void;
+  /** Alternative action callback */
+  onAlternative?(): void;
 }
 
-export const ConfirmModal: FC<Props> = ({
+export const ConfirmModal = ({
   isOpen,
   title,
   body,
+  description,
   confirmText,
+  confirmationText,
   dismissText = 'Cancel',
+  alternativeText,
   icon = 'exclamation-triangle',
   onConfirm,
   onDismiss,
-}) => {
-  const theme = useContext(ThemeContext);
-  const styles = getStyles(theme);
+  onAlternative,
+}: ConfirmModalProps): JSX.Element => {
+  const [disabled, setDisabled] = useState<boolean>(Boolean(confirmationText));
+  const styles = useStyles(getStyles);
+  const onConfirmationTextChange = (event: React.FormEvent<HTMLInputElement>) => {
+    setDisabled(confirmationText?.localeCompare(event.currentTarget.value, undefined, { sensitivity: 'accent' }) !== 0);
```

Comment:
Just wondering - do we need this localeCompare complexity here? Guess `===` would do it's job too :) Especially that typing "Delete" and "delete" makes it pass - don't know if this is a regression, probably not as we had `scope.confirmTextValid = payload.confirmText.toLowerCase() === value.toLowerCase();` before.

I think tho, that if the confitrm text is "Delete" then the input from the user should respect capitalization of D, as a matter of performing dangerous action

Reply:
I agree with your point that we should keep the exact capitalization. I think I'll change that. Regarding the `localeCompare` I think that is the most correct way of comparing texts according to this https://github.com/grafana/grafana/pull/32435#issuecomment-809934792 
https://stackoverflow.com/questions/2140627/how-to-do-case-insensitive-string-comparison/51005583#51005583

Reply:
Made it so we need the exact confirmation text instead, but still using localeCompare :)